packages/features/bookings/lib/getAllCredentialsForUsersOnEvent/getAllCredentials.ts

```diff
@@ -72,5 +75,44 @@ export const getAllCredentials = async (
     }
   }
 
+  // Only return CRM credentials that are enabled on the event type
+  const eventTypeAppMetadata = eventType?.metadata?.apps;
+
+  // Will be [credentialId]: { enabled: boolean }]
+  const eventTypeCrmCredentials: Record<number, { enabled: boolean }> = {};
+
+  for (const appKey in eventTypeAppMetadata) {
+    const app = eventTypeAppMetadata[appKey as keyof typeof eventTypeAppMetadata];
+    if (app.appCategories && app.appCategories.some((category: string) => category === "crm")) {
+      eventTypeCrmCredentials[app.credentialId] = {
+        enabled: app.enabled,
+      };
+    }
+  }
+
+  allCredentials = allCredentials.filter((credential) => {
+    if (!credential.type.includes("_crm") && !credential.type.includes("_other_calendar")) {
+      return credential;
+    }
+
+    // Backwards compatibility: All CRM apps are triggered for every event type. Unless disabled on the event type
+    // Check if the CRM app exists on the event type
+    if (eventTypeCrmCredentials[credential.id]) {
```

Comment:
Following code could be abstracted out in something like `filterOutDisabledCrmCredentials`. Also that makes me realize, why do we check for CRM only. Shouldn't we check for any app here.

We could assume that if `enabled` is undefined it is actually true otherwise whatever it says.

Reply:
Payment apps are handled in a different function. Ideally that logic should live here right?

Reply:
Yeah. If it is similar logic it should be together only.